src/cephadm/cephadmlib/daemons/ceph.py

```diff
@@ -328,8 +328,10 @@ def get_daemon_args(self) -> List[str]:
         return args
 
     def validate(self) -> None:
+        if not os.path.exists(self.sock_dir):
```

Comment:
It feels a bit weird to create stuff in a func called validate().
In the smb pr I was trying to add a new place for a service to create dirs and files it needs - see https://github.com/ceph/ceph/pull/55068/commits/0f6a18a09cba35515e100f457e224537ad1e5608 and how it's used by the new servcie.

Would that be more appropriate for this?

Reply:
pulled your patches in and rewrote this to use `prepare_data_dir`